templates/default/components/Layout.jsx

```diff
@@ -1,13 +1,76 @@
 import Head from "next/head";
 import Link from "next/link";
+import { useCallback, useEffect, useState } from "react";
 
 import { siteConfig } from "../config/siteConfig";
 import { Nav } from "./Nav";
 
-export function Layout({ children }) {
+function useTableOfContents(tableOfContents) {
+  const [currentSection, setCurrentSection] = useState(tableOfContents[0]?.id);
+
+  const getHeadings = useCallback((toc) => {
+    return toc
+      .flatMap((node) => [node.id, ...node.children.map((child) => child.id)])
+      .map((id) => {
+        const el = document.getElementById(id);
+        if (!el) return;
+
+        const style = window.getComputedStyle(el);
+        const scrollMt = parseFloat(style.scrollMarginTop);
+
+        const top = window.scrollY + el.getBoundingClientRect().top - scrollMt;
+        return { id, top };
+      });
+  }, []);
+
+  useEffect(() => {
+    if (tableOfContents.length === 0) return;
+    const headings = getHeadings(tableOfContents);
+    function onScroll() {
+      const top = window.scrollY + 4.5;
+      let current = headings[0].id;
+      // for (const heading of headings) {
+      headings.map((heading) => {
+        if (top >= heading.top) {
+          current = heading.id;
+        }
+        return current;
+      });
+      // }
+      setCurrentSection(current);
+    }
+    window.addEventListener("scroll", onScroll, { passive: true });
+    onScroll();
+    return () => {
+      window.removeEventListener("scroll", onScroll, { passive: true });
+    };
+  }, [getHeadings, tableOfContents]);
+
+  return currentSection;
+}
+
+export function Layout({ children, tableOfContents }) {
   const { editLink, _raw } = children.props;
   /* if editLink is not set in page frontmatter, link bool value will depend on siteConfig.editLinkShow */
   const editUrl = `${siteConfig.repoRoot}${siteConfig.repoEditPath}${_raw?.sourceFilePath}`;
+
+  const showToc = Array.isArray(siteConfig.tableOfContents)
+    ? siteConfig.tableOfContents.includes(_raw?.sourceFileDir) ||
+      siteConfig.tableOfContents.includes(_raw?.flattenedPath)
+    : _raw?.flattenedPath;
```

Comment:
So `showToc` can either be a boolean or a string ? (_raw?.flattenedPath). I think I don't understand :)

Reply:
_raw?.flattenedPath for index.md (ie. frontpage) returns an empty string and because of this the ToC wont render on the homepage. It's not a boolean but being used like one.

Reply:
Yes, I figured that you're using this string value as a truthy/falsy value but it's hard to understand what you're trying to achieve for someone who doesn't have the explanation. Also, if the `siteConfig.tableOfContents` is not an array, you're completely ignoring it's boolean value which can be either `true` or `false`, because all the pages apart from the index page will not be an empty string. So `showToc` can evaluate to the truth value even though you have `tableOfContents` set to `false` in the config.

Reply:
Right will make those changes 👍